storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -353,25 +350,21 @@ public LocalState nimbusTopoHistoryStateImpl(Map conf) throws IOException {
     }
 
     // we use this "weird" wrapper pattern temporarily for mocking in clojure test
-    public static Map readSupervisorStormConf(Map conf, String stormId) throws IOException {
+    public static Map<String, Object> readSupervisorStormConf(Map<String, Object> conf, String stormId) throws IOException {
         return _instance.readSupervisorStormConfImpl(conf, stormId);
     }
 
-    public Map readSupervisorStormConfImpl(Map conf, String stormId) throws IOException {
+    public Map<String, Object> readSupervisorStormConfImpl(Map<String, Object> conf, String stormId) throws IOException {
         String stormRoot = supervisorStormDistRoot(conf, stormId);
         String confPath = supervisorStormConfPath(stormRoot);
         return readSupervisorStormConfGivenPath(conf, confPath);
     }
 
     // we use this "weird" wrapper pattern temporarily for mocking in clojure test
-    public static StormTopology readSupervisorTopology(Map conf, String stormId) throws IOException {
-        return _instance.readSupervisorTopologyImpl(conf, stormId);
-    }
-
-    public StormTopology readSupervisorTopologyImpl(Map conf, String stormId) throws IOException {
+    public static StormTopology readSupervisorTopology(Map conf, String stormId, AdvancedFSOps ops) throws IOException {
```

Comment:
I put in the AdvancedFSOps to handle the mocking, but now I am not so sure on that.  It is too low level for a lot of what we want to do, so I think I will put it back in, and just do the mocking at the higher level.